tests/CMakeLists.txt

```diff
@@ -5,9 +5,22 @@
 # Mozilla Public License Version 2.0
 
 # Project include(s).
-include( vecmem-googletest )
 include( vecmem-compiler-options-cpp )
 
+# Set up GoogleTest.
+option( VECMEM_USE_SYSTEM_GOOGLETEST
+   "Pick up an existing installation of GoogleTest from the build environment"
+   FALSE )
+if( VECMEM_USE_SYSTEM_GOOGLETEST )
+   find_package( GTest )
+   # If GoogleTest is not found, don't fail right away. It might be that we are
+   # building VecMem as part of another project, which takes care of building
+   # GoogleTest itself. CMake will fail anyway if the GTest:: targets are not
+   # found by the end of the project's configuration.
```

Comment:
I'm not too sure about this. There is always a dependency chain, and some node has no dependencies (correct me if I'm wrong). So in my opinion the parent project just has to make sure the dependencies are set up in a way that the chain works. In that case, we should just fail here.

Reply:
Consider a directory layout like:

```
parent_project
|- vecmem
|- project_providing_googletest
```

In this setup the `GTest::gtest` and `GTest::gtest_main` targets will only be defined in `project_providing_googletest`, and are not yet visible in vecmem. So we can't even check with `if( TARGET "GTest::gtest" )` for this.

No, this logic that I wrote down does not work for all situations after all... :frowning:

Reply:
And

```
parent_project
|- project_providing_googletest
|- vecmem
``` 

doesn't work either because of the directory scoping?

Reply:
No, that could potentially work. Since if `project_providing_googletest` builds GoogleTest, it has to create "global" targets for it.

But having to carefully manage include orders like this does not sound like fun. And in our code we would still need to use `if( TARGET "GTest::gtest" )`. Which is also pretty fragile.

BTW this is the first time that I saw discussions showing up in VSCode. :stuck_out_tongue:

![Screenshot 2021-09-30 10:52:02](https://user-images.githubusercontent.com/30694331/135420792-057a58ee-90c2-49ab-9cb7-f095713936e6.png)